File: mobile/src/main/java/com/ataulm/whatsnext/letterboxd/ApiFilmSummary.java
package com.ataulm.whatsnext.letterboxd;

import com.google.gson.annotations.SerializedName;

class ApiFilmSummary {

    @SerializedName("id")
    String letterboxId;

    @SerializedName("name")
    String name;

    @SerializedName("releaseYear")
    int releaseYear;

    @SerializedName("poster")
    ApiImage poster;
}

File: letterboxd/src/main/java/com/ataulm/letterboxd/ApiFilmSummary.kt
package com.ataulm.letterboxd

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ApiFilmSummary(
    @field:Json(name = "name")
    val name: String,

    @field:Json(name = "releaseYear")
    val releaseYear: Int?,

    @field:Json(name = "directors")
    val directors: List<ApiContributor>?,

    @field:Json(name = "poster")
    val poster: ApiImage?,

    @field:Json(name = "links")
    val links: List<ApiLink>
)
